backend/src/routes/admin-routes.ts

```diff
@@ -76,50 +92,76 @@ const adminRoutes: FastifyPluginAsync = async (server) => {
 
       if (!/^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*-_&])[A-Za-z\d@$!%*-_&]{8,}$/.test(body.apiKey))
-        return reply.code(ReturnCode.FAILURE).send({ error: ErrorMessage.API_KEY_VALIDATION_FAILED })
+        return reply.code(ReturnCode.FAILURE).send({ error: ErrorMessage.API_KEY_VALIDATION_FAILED });
 
-      const wallet = new ethers.Wallet(body.privateKey);
+      const mnemonic = ethers.utils.entropyToMnemonic(
+        ethers.utils.randomBytes(16)
+      );
+      const wallet = ethers.Wallet.fromMnemonic(mnemonic);
+      const privateKey = wallet.privateKey;
       const publicAddress = await wallet.getAddress();
 
-      // Use Sequelize to find the API key
-      const result = await server.apiKeyRepository.findOneByWalletAddress(publicAddress);
+      if(!unsafeMode) {
+        const { 'x-signature': signature, 'x-timestamp': timestamp } = request.headers as IncomingHttpHeaders & AuthDto;
+        if(!signature || !timestamp)
+          return reply.code(ReturnCode.NOT_AUTHORIZED).send({ error: ErrorMessage.INVALID_SIGNATURE_OR_TIMESTAMP });
+        if(!verifySignature(signature, request.body as string, timestamp, server.config.HMAC_SECRET))
+          return reply.code(ReturnCode.NOT_AUTHORIZED).send({ error: ErrorMessage.INVALID_SIGNATURE_OR_TIMESTAMP });
 
-      if (result) {
-        request.log.error('Duplicate record found');
-        return reply.code(ReturnCode.FAILURE).send({ error: ErrorMessage.DUPLICATE_RECORD });
-      }
+        const command = new GetSecretValueCommand({SecretId: prefixSecretId + body.apiKey})
+        const secrets = await client.send(command).catch((err) => err);
 
-      await server.apiKeyRepository.create({
-        apiKey: body.apiKey,
-        walletAddress: publicAddress,
-        privateKey: encode(body.privateKey, server.config.HMAC_SECRET),
-        supportedNetworks: body.supportedNetworks,
-        erc20Paymasters: body.erc20Paymasters,
-        multiTokenPaymasters: body.multiTokenPaymasters ?? null,
-        multiTokenOracles: body.multiTokenOracles ?? null,
-        sponsorName: body.sponsorName ?? null,
-        logoUrl: body.logoUrl ?? null,
-        transactionLimit: body.transactionLimit ?? 0,
-        noOfTransactionsInAMonth: body.noOfTransactionsInAMonth ?? 10,
-        indexerEndpoint: body.indexerEndpoint ?? process.env.DEFAULT_INDEXER_ENDPOINT,
-        bundlerApiKey: body.bundlerApiKey ?? null,
-      });
+        if(!(secrets instanceof Error)) {
+          request.log.error('Duplicate record found');
+          return reply.code(ReturnCode.FAILURE).send({ error: ErrorMessage.DUPLICATE_RECORD });
+        }
+
+        const createCommand = new CreateSecretCommand({
+          Name: prefixSecretId + body.apiKey,
+          SecretString: JSON.stringify({
+            PRIVATE_KEY: privateKey,
+            PUBLIC_ADDRESS: publicAddress,
+            MNEMONIC: mnemonic
+          }),
+        });
+
```

Comment:
We need to also create a record on the connected db since the sponsorship olivies are handled by that so even in unsafeMode as false create one record on was secrets as above and also include database record creation for the same which also includes addition of policy record by default for the same apiKey

Reply:
@vignesha22 does this mean we have to delete the record also on /deleteKey endpoint ?